src/components/Dialog.tsx

```diff
@@ -10,27 +16,111 @@ interface DialogProps {
 
 interface DialogState {
   imgix: any; // TODO - replace with class type
+  isOpen: boolean;
+  allSources: Array<SourceProps>;
+  selectedSource: Partial<SourceProps>;
 }
 
+type SourceProps = {
+  id: string;
+  name: string;
+};
+
 export default class Dialog extends Component<DialogProps, DialogState> {
   constructor(props: DialogProps) {
     super(props);
 
-    const apiKey = (props as any).sdk?.parameters.installation.imgixAPIKey || "";
+    const apiKey =
+      (props as any).sdk?.parameters.installation.imgixAPIKey || '';
```

Comment:
Nit: is it possible to remove the `as any` here?

Reply:
Yes I believe so, it requires importing the necessary type from [ConfigScreen.tsx](https://github.com/imgix/contentful/blob/main/src/components/ConfigScreen.tsx) and merging it with `DialogProps`, I'm currently looking into it.